src/nlp/arrow_dataset.py

```diff
@@ -766,6 +774,7 @@ def map(
         self,
         function,
         with_indices: bool = False,
+        input_column: Optional[str] = None,
```

Comment:
You need to add `input_column` to the `cache_kwargs` of `map`.
Same thing for `fn_kwargs`

Reply:
E.g. dataset.map(tokenizer, input_columns=['sentence1', 'sentence2'])

Reply:
Good call. Added.